be1-go/hub/standard_hub/message_handling.go

```diff
@@ -590,10 +574,8 @@ func (h *Hub) handleReceivedMessage(socket socket.Socket, messageData message.Me
 		return xerrors.Errorf(publishError, err)
 	}
 
-	h.Lock()
 	h.hubInbox.StoreMessage(publish.Params.Message)
-	h.addMessageId(publish.Params.Channel, publish.Params.Message.MessageID)
-	h.Unlock()
+	h.messageIdsByChannel.Add(publish.Params.Channel, publish.Params.Message.MessageID)
```

Comment:
just wondering (it wasn't better before) - should these two things be updated atomically or is the software going to be okay if there's, say, 5 seconds between one and the other happening ?

Reply:
They should be updated atomically indeed. As a matter of fact, when I was rereading the Inbox code and usage, I realized that we had two different Inbox instances in the hub that essentially serve the same purpose. I think I can do some refactoring in that regards (by creating a new HubInbox type that embeds Inbox but has more features (Channels and MessagesById that only the Hub needs) in another PR.